cli/src/main/java/com/devonfw/tools/ide/tool/vscode/Vscode.java

```diff
@@ -28,4 +42,127 @@ protected String getBinaryName() {
     return "code";
   }
 
+  @Override
+  public void installPlugin(PluginDescriptor plugin) {
+
+  }
+
+  @Override
+  protected void installPlugins(Collection<PluginDescriptor> plugins) {
+
+    List<PluginDescriptor> pluginsToInstall = new ArrayList<>();
+    List<String> pluginsToRecommend = new ArrayList<>();
+
+    for (PluginDescriptor plugin : plugins) {
+      if (plugin.isActive()) {
+        pluginsToInstall.add(plugin);
+      } else {
+        pluginsToRecommend.add(plugin.getId());
+      }
+    }
+    doAddRecommendations(pluginsToRecommend.toArray(new String[0]));
+    doInstallPlugins(pluginsToInstall);
+
+  }
+
+  private void doInstallPlugins(List<PluginDescriptor> pluginsToInstall) {
+    if (pluginsToInstall.isEmpty()) {
+      this.context.info("No plugins to be installed");
+    } else {
+      for (PluginDescriptor plugin : pluginsToInstall) {
+        this.arguments.addValue("--install-extension");
+        this.arguments.addValue(plugin.getId());
+      }
```

Comment:
You need to call `runTool` here with the arguments passed as `args` parameter to that method.
Otherwise you are causing a bug here:
* Assume the user will call `ide vscode arg1 arg2`.
* The `Vscode` commandlet gets `arg1` and `arg2` assigned to `this.arguments` property.
* Now if VSCode is not yet installed it will be installed and during that this method is called to install the plugins.
* Here you will add `--install-extensions`, etc. to `this.arguments` after `arg1` and `arg2`.

Instead we want to call `code` here via `runTool` to install the plugins now. If that fails, the installation fails and the actual run of `VSCode` intended by the end-user shall not even be triggered.
I hope it gets clear now.

Reply:
@aBega2000 can you please fix this review comment? We already discussed that you missed this one and it is trivial to fix but currently this is causing a severe bug.

Reply:
Thanks for your update. Well done 👍